pkg/cloudprovider/provider/kubevirt/provider.go

```diff
@@ -159,12 +176,19 @@ func (p *provider) Get(machine *v1alpha1.Machine, _ *cloudprovidertypes.Provider
 	}
 	ctx := context.Background()
 
+	virtualMachine := &kubevirtv1.VirtualMachine{}
+	if err := sigClient.Get(ctx, types.NamespacedName{Namespace: c.Namespace, Name: machine.Name}, virtualMachine); err != nil {
+		if !kerrors.IsNotFound(err) {
+			return nil, fmt.Errorf("failed to get VirtualMachine %s: %v", machine.Name, err)
+		}
+		return nil, cloudprovidererrors.ErrInstanceNotFound
+	}
+
 	virtualMachineInstance := &kubevirtv1.VirtualMachineInstance{}
 	if err := sigClient.Get(ctx, types.NamespacedName{Namespace: c.Namespace, Name: machine.Name}, virtualMachineInstance); err != nil {
 		if !kerrors.IsNotFound(err) {
-			return nil, fmt.Errorf("failed to get VirtualMachineInstance %s: %v", machine.Name, err)
+			return &kubeVirtServer{}, nil
```

Comment:
Shouldn't we return the error here? Why should we swallow errors that are not 404?

IMHO we should return the empty kubevirtServer and a nil error if the error _is_ a 404

Reply:
I fully agree, my bad